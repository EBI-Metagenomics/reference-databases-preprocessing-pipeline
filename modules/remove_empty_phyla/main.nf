
process REMOVE_EMPTY_PHYLA {

    label 'light'
    publishDir "${params.outdir}/${label}/${version}/", mode: 'copy'
    
    input:
    path taxid
    val version
    val label

    output:
    path ("*-tax.txt"), emit: tax

    """
    grep -v "p__;" $taxid > ${label}-tax.txt
    """

}