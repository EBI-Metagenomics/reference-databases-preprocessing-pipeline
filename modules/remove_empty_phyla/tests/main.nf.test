nextflow_process {

    name "Test Process REMOVE_EMPTY_PHYLA"
    script "modules/remove_empty_phyla/main.nf"
    process "REMOVE_EMPTY_PHYLA"
    tag "modules"
    tag "remove_empty_phyla"

    test("Test removal of empty phyla process") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = file('${projectDir}/modules/remove_empty_phyla/tests/data/test-tax.txt', checkIfExists: true)
                input[1] = params.silva_version
                input[2] = params.silva_ssu_label
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.tax).match("tax") }
            )
        }

    }

}
