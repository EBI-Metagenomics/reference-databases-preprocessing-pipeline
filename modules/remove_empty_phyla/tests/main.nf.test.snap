{
    "tax": {
        "content": [
            [
                "SILVA-SSU-tax.txt:md5,1be080c83abb578f02e73ba516bad3f0"
            ]
        ],
        "timestamp": "2023-12-12T09:15:53.847331966"
    }
}